dff/messengers/telegram.py

```diff
@@ -0,0 +1,179 @@
+                    return
+                if isinstance(attachment, Poll):
+                    await bot.send_poll(chat_id, attachment.question, [option.text for option in attachment.options], reply_markup=self._create_keyboard(buttons))
+                    return
+                if isinstance(attachment, Audio):
+                    attachment_bytes = await attachment.get_bytes(self)
+                    if attachment_bytes is not None:
+                        files += [InputMediaAudio(attachment_bytes)]
+                if isinstance(attachment, Video):
+                    attachment_bytes = await attachment.get_bytes(self)
+                    if attachment_bytes is not None:
+                        files += [InputMediaVideo(attachment_bytes)]
+                if isinstance(attachment, Animation):
+                    attachment_bytes = await attachment.get_bytes(self)
+                    if attachment_bytes is not None:
+                        files += [InputMediaAnimation(attachment_bytes)]
+                if isinstance(attachment, Image):
+                    attachment_bytes = await attachment.get_bytes(self)
+                    if attachment_bytes is not None:
+                        files += [InputMediaPhoto(attachment_bytes)]
+                if isinstance(attachment, Document):
+                    attachment_bytes = await attachment.get_bytes(self)
+                    if attachment_bytes is not None:
+                        files += [InputMediaDocument(attachment_bytes)]
+                if isinstance(attachment, Keyboard):
+                    buttons = attachment.buttons
+            if len(files) > 0:
+                await bot.send_media_group(chat_id, files, caption=message.text)
+                return
+        if message.text is not None:
+            await bot.send_message(chat_id, message.text, reply_markup=self._create_keyboard(buttons))
+
+    async def _on_event(self, update: Update, _: ContextTypes.DEFAULT_TYPE, create_message: Callable[[Update], Message]) -> None:
+        if update.effective_chat is not None and update.message is not None:
+            message = create_message(update)
+            message.original_message = update
+            resp = await self.callback(message, update.effective_chat.id)
+            if resp.last_response is not None:
+                await self.cast_message_to_telegram_and_send(self.application.bot, update.effective_chat.id, resp.last_response)
+
+    async def on_message(self, update: Update, _: ContextTypes.DEFAULT_TYPE) -> None:
+        await self._on_event(update, _, lambda u: await self.extract_message_from_telegram(u.message))
+
+    async def on_callback(self, update: Update, _: ContextTypes.DEFAULT_TYPE) -> None:
+        await self._on_event(update, _, lambda u: Message(attachments=[CallbackQuery(query_string=u.callback_query.data)]))
+
+    async def connect(self, callback: PipelineRunnerFunction, *args, **kwargs):
+        self.callback = callback
```

Comment:
Rename it to `pipeline_runner`.
It creates confusion making it seem as if it is related to `on_callback`.

Reply:
This is not resolved.